distribution/src/config/jvm.options

```diff
@@ -76,16 +76,12 @@ ${error.file}
 # JDK 9+ GC logging
 9-:-Xlog:gc*,gc+age=trace,safepoint:file=${loggc}:utctime,pid,tags:filecount=32,filesize=64m
 
-# Explicitly allow security manager (https://bugs.openjdk.java.net/browse/JDK-8270380)
-18-:-Djava.security.manager=allow
-
 # JDK 20+ Incubating Vector Module for SIMD optimizations;
 # disabling may reduce performance on vector optimized lucene
 20-:--add-modules=jdk.incubator.vector
 
-# HDFS ForkJoinPool.common() support by SecurityManager
--Djava.util.concurrent.ForkJoinPool.common.threadFactory=org.opensearch.secure_sm.SecuredForkJoinWorkerThreadFactory
-
 # See please https://bugs.openjdk.org/browse/JDK-8341127 (openjdk/jdk#21283)
 23:-XX:CompileCommand=dontinline,java/lang/invoke/MethodHandle.setAsTypeCache
 23:-XX:CompileCommand=dontinline,java/lang/invoke/MethodHandle.asTypeUncached
+
+21-:-javaagent:agent/opensearch-agent.jar
```

Comment:
not having this included JVM options will still run the Opensearch process successfully (i.e without the Agent)? That would be useful for plugins who would want zero overhead of stack walking from interceptors.

Reply:
> I believe we have mandate the presence of the agent 

I think maybe we should "only" mandate the SystemExit and SystemHalt, the File & Socket interceptors can be kept optional. Although we haven't seen overhead of stackwalking with very deep stacks (100s), but maybe someone who places a socket call (an example) too-deep in their plugin code might see an overhead and would want to skip these interceptor. 

But don't think its a blocker for this PR. We should get this to merging and decide before 3.0 release.

Reply:
How should we proceed given that the beta1 RC generation is happening today?

How can we systematically identify the codebases/plugins that need new policy file entries?

Reply:
cc @peterzhuamazon

Reply:
Hi @kumargu @reta just to clarify you are talking about `rc2` for 3.0.0-beta1 right?
We will build rc1 without this change on 4/9 hopefully, and will onboard this change with rc2 on 4/10.

Thanks.

Reply:
> Hi @kumargu @reta just to clarify you are talking about `rc2` for 3.0.0-beta1 right? We will build rc1 without this change on 4/9 hopefully, and will onboard this change with rc2 on 4/10.
> 
> Thanks.

Yeah, exactly, I meant basically the "next beta release in the pipeline", thanks for correcting me @peterzhuamazon